packages/composer/test/composer.test.ts

```diff
@@ -0,0 +1,153 @@
+import { OperationResult, Operation } from '@urql/core'
+import { MetadataComposer } from '../src'
+import { merge } from '../src/MetadataComposer'
+import { Source, parse } from 'graphql'
+
+const urls = [
+  'https://api.thegraph.com/subgraphs/name/yuetloo/metadata-rinkeby',
+]
+
+function createOperation(key: number): Operation<any, any> {
+  const source = new Source('{ permissions{id} }')
+  const document = parse(source)
+
+  return {
+    kind: 'query',
+    context: { url: 'x', requestPolicy: 'cache-first' },
+    key,
+    query: document,
+  }
+}
+
+function createMetadata(key: number) {
+  return {
+    id: `key${key}`,
+    metadata: `metadata${key}`,
+    owner: `owner${key}`,
+  }
+}
+
+function createPermission(key: number) {
+  return {
+    id: `p${key}`,
+  }
+}
+
+describe('MetadataComposer', () => {
```

Comment:
create issue to use GitHub Actions to Run Automated Tests on PR submission

Reply:
created issue #10 for this.